In [1]:
import numpy as np
import pandas as pd

In [2]:
airplane = pd.read_csv(r"C:\Users\user\Desktop\PYTHON PROGRAMING\BASIC PYTHON\EXTRAS\Airplane_Crashes_and_Fatalities_Since_1908.csv")

In [3]:
airplane.shape

(5268, 13)

In [4]:
airplane.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5268 entries, 0 to 5267
Data columns (total 13 columns):
Date            5268 non-null object
Time            3049 non-null object
Location        5248 non-null object
Operator        5250 non-null object
Flight #        1069 non-null object
Route           3562 non-null object
Type            5241 non-null object
Registration    4933 non-null object
cn/In           4040 non-null object
Aboard          5246 non-null float64
Fatalities      5256 non-null float64
Ground          5246 non-null float64
Summary         4878 non-null object
dtypes: float64(3), object(10)
memory usage: 329.3+ KB


In [5]:
airplane.columns

Index(['Date', 'Time', 'Location', 'Operator', 'Flight #', 'Route', 'Type',
       'Registration', 'cn/In', 'Aboard', 'Fatalities', 'Ground', 'Summary'],
      dtype='object')

In [6]:
airplane.columns = ['date', 'time', 'location', 'operator', 'flight', 'route', 'type',
       'registration', 'cn_in', 'aboard', 'fatalities', 'ground', 'summary']
airplane.columns

Index(['date', 'time', 'location', 'operator', 'flight', 'route', 'type',
       'registration', 'cn_in', 'aboard', 'fatalities', 'ground', 'summary'],
      dtype='object')

The information below shows that the flight column has approximately 80% null values
there is need for us to remove the column

In [7]:
(airplane.isnull().sum() / airplane.shape[0]) * 100

date             0.000000
time            42.122248
location         0.379651
operator         0.341686
flight          79.707669
route           32.384207
type             0.512528
registration     6.359150
cn_in           23.310554
aboard           0.417616
fatalities       0.227790
ground           0.417616
summary          7.403189
dtype: float64

In [8]:
airplane = airplane.drop(["flight"], axis=1)
airplane.shape # the flight column is removed and we are left with 12 columns

(5268, 12)

check the ground column to see if it only contains 0 and 1. If so convert it to a Boolean type

since there are other numbers aside 1 and 0 we leave it like that 

In [9]:
airplane["ground"].value_counts()

0.0       5027
1.0         53
2.0         27
3.0         21
4.0         15
5.0         10
8.0         10
7.0          8
11.0         6
6.0          6
22.0         5
13.0         4
20.0         3
10.0         3
44.0         3
24.0         3
52.0         2
30.0         2
14.0         2
12.0         2
2750.0       2
70.0         2
19.0         2
47.0         2
37.0         1
31.0         1
35.0         1
49.0         1
75.0         1
63.0         1
125.0        1
23.0         1
50.0         1
36.0         1
16.0         1
54.0         1
17.0         1
113.0        1
33.0         1
58.0         1
225.0        1
107.0        1
45.0         1
32.0         1
40.0         1
9.0          1
87.0         1
18.0         1
85.0         1
15.0         1
Name: ground, dtype: int64

 Below is the Average plane crashes  per year

In [10]:
airplane["date"] = pd.to_datetime(airplane["date"])
airplane["date"]


0      1908-09-17
1      1912-07-12
2      1913-08-06
3      1913-09-09
4      1913-10-17
          ...    
5263   2009-05-20
5264   2009-05-26
5265   2009-06-01
5266   2009-06-07
5267   2009-06-08
Name: date, Length: 5268, dtype: datetime64[ns]

In [11]:
airplane["year"] = airplane["date"].dt.year
airplane["year"]

0       1908
1       1912
2       1913
3       1913
4       1913
        ... 
5263    2009
5264    2009
5265    2009
5266    2009
5267    2009
Name: year, Length: 5268, dtype: int64

In [12]:
airplane["year"].value_counts().mean()

53.755102040816325

In [13]:
# with the least/most crashes?
airplane["year"].value_counts()

1972    104
1968     96
1989     95
1967     91
1973     89
       ... 
1918      4
1913      3
1915      2
1912      1
1908      1
Name: year, Length: 98, dtype: int64

In [14]:
airplane["year"].value_counts().max()

104

In [15]:
airplane["year"].value_counts().min()

1

# least and most
the output above shows that 1912 and 1908 have the least plane crashes while 1972 has the highest plane crashes.

In [16]:
airplane['month_difference'] = (airplane['date'] - airplane.shift(1)['date']) # create a new column
airplane['month_difference']

0            NaT
1      1394 days
2       390 days
3        34 days
4        38 days
          ...   
5263     17 days
5264      6 days
5265      6 days
5266      6 days
5267      1 days
Name: month_difference, Length: 5268, dtype: timedelta64[ns]

In [17]:
airplane['month_difference'].mean()

Timedelta('6 days 23:38:07.677995')

In [18]:
airplane['month_difference'].describe()

count                        5267
mean       6 days 23:38:07.677995
std      310 days 19:57:28.381069
min          -7003 days +00:00:00
25%               1 days 00:00:00
50%               4 days 00:00:00
75%               8 days 00:00:00
max            7020 days 00:00:00
Name: month_difference, dtype: object

The output above shows that the longest period inbetween crashes is 7020 days

In [19]:
airplane["operator"].value_counts(ascending=False)

Aeroflot                        179
Military - U.S. Air Force       176
Air France                       70
Deutsche Lufthansa               65
Air Taxi                         44
                               ... 
Bruns Air Service - Air Taxi      1
Tassili Airlines                  1
VASP / Private                    1
Wyoming Air Service               1
Sky Executive Air Services        1
Name: operator, Length: 2476, dtype: int64

The output above shows that Aeroflot suufered the most plane crash in history

In [20]:
airplane['survival_rate'] = airplane['aboard'] - airplane['fatalities']
airplane['survival_rate']

0        1.0
1        0.0
2        0.0
3        6.0
4        0.0
        ... 
5263    14.0
5264     0.0
5265     0.0
5266     0.0
5267     0.0
Name: survival_rate, Length: 5268, dtype: float64

In [21]:
airplane['survival_rate'].mean()

7.471025543271064

The output shows the average survival rate